### PR TITLE
Update consumer add-on install rosa command for additional parameter 

### PR DESCRIPTION
**Pull request by suchita-g**
Update rosa add-on install command for onbording ticket
 and storage-provider-endpoint parameters

Signed-off-by: suchita.gatfane <sgatfane@redhat.com>

### CODE REVIEW

File: ocs_ci/utility/utils.py [hunk 1]
```diff
@@ -3636,3 +3636,4 @@ def decode(encoded_message):
     decoded_base64_bytes = base64.b64decode(encoded_message_bytes)
     decoded_message = decoded_base64_bytes.decode("ascii")
     return decoded_message
+
```
Comment (fbalak) [inline ocs_ci/utility/utils.py:3639]: This blank line fails flake8 check.

**Review by fbalak — COMMENTED** [review]
LGTM, just flake8 error should be handled.

**Review by fbalak — DISMISSED** [review]

File: ocs_ci/utility/rosa.py [hunk 2]
```diff
@@ -230,6 +237,16 @@ def install_odf_addon(cluster):
         public_key_only = remove_header_footer_from_key(public_key)
         cmd += f' --onboarding-validation-key "{public_key_only}"'
 
+    if cluster_type.lower() == "consumer" and config.ENV_DATA.get("provider_name", ""):
```
Comment (yitzhak12) [inline ocs_ci/utility/rosa.py:240]: You already got the provider name in line 215. you can replace `config.ENV_DATA.get("provider_name", "")` with the var  `provider_name`

**Review by yitzhak12 — COMMENTED** [review]
Other than my one comment, the pr looks good.

**Review by fbalak — APPROVED** [review]

**Review by jilju — DISMISSED** [review]

File: ocs_ci/utility/managedservice.py [hunk 3]
```diff
@@ -90,3 +90,31 @@ def remove_header_footer_from_key(key):
...
+    #  handle invalid cluster name in implementation
+    #  validate Return String storage provider endpoint:
+    #     1. raise Error if storage_provider_endpoint is
+    #        not found in cluster yaml
+    #     2. warning if storage cluster is not ready
+    #     and storage_provider_endpoint is available in
+    #     storagecluster yaml .
+    #  For now use hardcoded value from config with key
+    #  storage_provider_endpoint:
+    return config.Deployment.get("storage_provider_endpoint", "")
```
Comment (dahorak) [inline ocs_ci/utility/managedservice.py:120]:
```suggestion
    return config.DEPLOYMENT.get("storage_provider_endpoint", "")
```

**Review by jilju — APPROVED** [review]

**Review by fbalak — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Review by fbalak — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**